modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java

```diff
@@ -253,10 +260,29 @@ protected void activate(ComponentContext cc) {
     ltiRolesForUserCreation = extractLtiRolesForUserCreation(
             Objects.toString(properties.get(LTI_ROLES_TO_CREATE_JPA_USER_REFERENCES_FROM), "*"));
 
-    customRoleName = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLE_NAME));
+    for (int i = 1; true; i++) {
+      logger.debug("Looking for custom role configuration of {}", CUSTOM_ROLE_NAME_PREFIX + i);
+      String customRoleName = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLE_NAME_PREFIX + i));
+      if (customRoleName == null) {
+        break;
+      }
+      String customRolesString = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLES_LIST_PREFIX + i));
+      if (customRolesString == null) {
+        continue;
+      }
+      String[] customRoles = customRolesString.split(",");
+      Pattern customRolePattern = Pattern.compile(customRoleName);
+      customRolePatterns.put(customRolePattern, customRoles);
+    }
+
+    // Add support for legacy custom_role_name
+    String customRoleName = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLE_NAME));
+
     if (customRoleName != null) {
-      String custumRolesString = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLES));
-      customRoles = custumRolesString.split(",");
+      String customRolesString = StringUtils.trimToNull((String) properties.get(CUSTOM_ROLES));
+      String[] customRoles = customRolesString.split(",");
+      Pattern customRolePattern = Pattern.compile(customRoleName);
+      customRolePatterns.put(customRolePattern, customRoles);
```

Comment:
Little bit of code duplication going on here, same code can be found in lines 273-275. Might be worth turning this into a function?

Reply:
Hi sorry for not answering sooner didnt check my github activity, one question about release notes I see that there's no Opencast 15 version heading, should I add one?

Reply:
No, you don't need to. Just add your own `changes-to-lti-configuration.txt` to the folder, with a short note on what changed. The release managers will take care of proper formatting when OC15 releases.